source/mpi_shared_solid_solver.cpp

```diff
@@ -162,6 +162,14 @@ namespace Solid
           spacedim,
           PETScWrappers::MPI::Vector(locally_owned_scalar_dofs,
                                      mpi_communicator)));
+
+      cellwise_stress = std::vector<Vector<double>>(
+        6, Vector<double>(triangulation.n_active_cells()));
```

Comment:
Here comes the problem: when the problem is 2D, the stress only has 3 components. So the size of ``cellwise_stress`` should be dependent on ``spacedim``, not fixed to 6. Line 159-164 can be a good example of this.